plugins/pencilblue/templates/angular/admin/content/objects/object_form.html

```diff
@@ -1,251 +1,210 @@
-            }
-        };
-
-        for(var i = 0; i < $scope.objectType.fields.length; i++) {
-            if($scope.objectType.fields[i].field_type === 'boolean' && typeof $scope.customObject[$scope.objectType.fields[i].name] === 'undefined') {
-                $scope.customObject[$scope.objectType.fields[i].name] = true;
-            }
-            else if($scope.objectType.fields[i].field_type === 'date') {
-                $scope.customObject[$scope.objectType.fields[i].name] = $filter('date')($scope.customObject[$scope.objectType.fields[i].name], 'MM-dd-yyyy HH:mm');
-            }
-            else if($scope.objectType.fields[i].field_type === 'peer_object') {
-                $scope.customObject[$scope.objectType.fields[i].name] = $scope.customObject[$scope.objectType.fields[i].name] ? $scope.customObject[$scope.objectType.fields[i].name]._id.toString() : '';
-            }
-            else if($scope.objectType.fields[i].field_type === 'child_objects') {
-                $scope.objectType.fields[i].searchText = '';
-                $scope.objectType.fields[i].paginationIndex = 0;
-                $scope.objectType.fields[i].paginationLimit = 12;
-                $scope.paginate($scope.objectType.fields[i], 0);
-            }
+        index++;
+        self.loadWysiwyg(index);
+      });
+    };
+
+    this.loadWysiwyg(0);
+  };
+
+  for(var i = 0; i < $scope.objectType.fields.length; i++) {
+    if($scope.objectType.fields[i].field_type === 'boolean' && typeof $scope.customObject[$scope.objectType.fields[i].name] === 'undefined') {
+      $scope.customObject[$scope.objectType.fields[i].name] = true;
+    }
+    else if($scope.objectType.fields[i].field_type === 'date') {
+      $scope.customObject[$scope.objectType.fields[i].name] = $filter('date')($scope.customObject[$scope.objectType.fields[i].name], 'MM-dd-yyyy HH:mm');
+    }
+    else if($scope.objectType.fields[i].field_type === 'peer_object') {
+      $scope.customObject[$scope.objectType.fields[i].name] = $scope.customObject[$scope.objectType.fields[i].name] ? $scope.customObject[$scope.objectType.fields[i].name]._id.toString() : '';
+    }
+    else if($scope.objectType.fields[i].field_type === 'child_objects') {
+      $scope.objectType.fields[i].searchText = '';
+      $scope.objectType.fields[i].paginationIndex = 0;
+      $scope.objectType.fields[i].paginationLimit = 12;
+      $scope.paginate($scope.objectType.fields[i], 0);
+    }
+  }
+})
+.directive('fieldsBuilt', function() {
+  return function(scope, element, attrs) {
+    var self = this;
+
+    this.testForDateFields = function() {
+      if(!$('.date_field').length) {
+        setTimeout(self.testForDateFields, 100);
+        return;
+      }
+
+      $('.date_field').datetimepicker({format: 'm-d-Y H:i'});
+    }
+
+    if(scope.$last){
+      for(var i = 0; i < scope.objectType.fields.length; i++) {
+        if(scope.objectType.fields[i].field_type === 'date') {
+          this.testForDateFields();
         }
```

Comment:
No, that's what scope.$last is. It'll only run that code on the last run of
the ng-repeat.

On Mon, Jun 22, 2015 at 9:49 AM, Brian <notifications@github.com> wrote:

> In
> plugins/pencilblue/templates/angular/admin/content/objects/object_form.html
> <https://github.com/pencilblue/pencilblue/pull/660#discussion_r32935274>:
>
> > +  return function(scope, element, attrs) {
> > +    var self = this;
> > +
> > +    this.testForDateFields = function() {
> > +      if(!$('.date_field').length) {
> > +        setTimeout(self.testForDateFields, 100);
> > +        return;
> > +      }
> > +
> > +      $('.date_field').datetimepicker({format: 'm-d-Y H:i'});
> > +    }
> > +
> > +    if(scope.$last){
> > +      for(var i = 0; i < scope.objectType.fields.length; i++) {
> > +        if(scope.objectType.fields[i].field_type === 'date') {
> > +          this.testForDateFields();
> >          }
>
> ah, got it. Then do you need to break out of the loop after it makes the
> call?
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/pencilblue/pencilblue/pull/660/files#r32935274>.
>